apps/sms/views/conversation/js/conversation.js

```diff
@@ -2776,21 +2777,17 @@ var ConversationView = {
 
       this.prompt({
         number: tel,
-        email: email,
+        email,
         header: fragment,
         contactId: id,
-        isContact: isContact,
-        inMessage: inMessage
+        isContact,
+        inMessage: opts.inMessage
       });
     }.bind(this));
   },
 
   prompt: function conv_prompt(opt) {
-    var complete = (function complete() {
-      if (!Navigation.isCurrentPanel('thread')) {
-        Navigation.toPanel('thread', { id: Threads.currentId });
-      }
-    }).bind(this);
```

Comment:
this was good enough in a pre-promise world -- moved it to the right place: https://github.com/mozilla-b2g/gaia/pull/30630/files#diff-cda15883484103296e7c95b4f1512099R226